src/hotspot/share/gc/shenandoah/shenandoahHeap.inline.hpp

```diff
@@ -312,7 +312,13 @@ inline oop ShenandoahHeap::evacuate_object(oop p, Thread* thread) {
     // Successfully evacuated. Our copy is now the public one!
     shenandoah_assert_correct(NULL, copy_val);
 
-    // Increment age in young copies
+    // Hey!  This code showed up in a merge conflict.  It has "nothing" to do with the patch that
+    // was merged, so kdnilsen is leaving it in place as is.  However, it looks to me like the object's
+    // age should be incremented before the copy is committed to avoid the need for synchronization here.
+    //
+    // kdnilsen believes the following code is replaced/relocated in a subsequent commit.
+
+    // Increment age in young copies.
     if (target_gen == YOUNG_GENERATION) {
       copy_val->incr_age();
```

Comment:
Now, age is increased during young gen copy. Please update ShenandoahStringDedup::enqueue_candidate() to remove age updating code.

Reply:
We did discover the need to deal with displaced mark words and a subsequent commit replaces the call to incr_age() with a call to a newly introduced incrase_object_age() function which includes special handling for displaced mark words.

Reply:
I have also removed the code in ShenandoahStringDedup::enqueue_candidate() which increments the age of strings since we are now incrementing age during evacuation.